sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/XmlExternalEntityShouldNotBeParsed_XmlDocument.cs

```diff
@@ -0,0 +1,193 @@
+        protected void ConfigXmlDocumentTest()
+        {
+            ConfigXmlDocument doc = new ConfigXmlDocument();
+            doc.XmlResolver = new XmlPreloadedResolver(); // Noncompliant
+        }
+
+        // Microsoft.Web.XmlTransform.XmlFileInfoDocument
+        protected void XmlFileInfoDocumentTest()
+        {
+            XmlFileInfoDocument doc = new XmlFileInfoDocument();
+            doc.XmlResolver = new XmlPreloadedResolver(); // Noncompliant
+        }
+
+        // Microsoft.Web.XmlTransform.XmlTransformableDocument
+        protected void XmlTransformableDocumentTest()
+        {
+            XmlTransformableDocument doc = new XmlTransformableDocument();
+            doc.XmlResolver = new XmlPreloadedResolver(); // Noncompliant
+        }
+    }
+
+    class CompliantTests_After_Net_4_5_2
+    {
+        // System.Xml.XmlDocument
+        protected void XmlDocument_1(XmlSecureResolver xmlSecureResolver)
+        {
+            XmlDocument doc = new XmlDocument();
+            doc.XmlResolver = xmlSecureResolver;
+        }
+
+        protected void XmlDocument_2(XmlSecureResolver xmlSecureResolver)
+        {
+            XmlDocument doc = new XmlDocument();
+            doc.XmlResolver = null;
+        }
+
+        // System.Xml.XmlDataDocument
+        protected void XmlDataDocumentTest()
+        {
+            XmlDataDocument doc = new XmlDataDocument();
+        }
+
+        // System.Configuration.ConfigXmlDocument
+        protected void ConfigXmlDocumentTest()
+        {
+            ConfigXmlDocument doc = new ConfigXmlDocument();
+        }
+
+        // Microsoft.Web.XmlTransform.XmlFileInfoDocument
+        protected void XmlFileInfoDocumentTest()
+        {
+            XmlFileInfoDocument doc = new XmlFileInfoDocument();
+        }
+
+        // Microsoft.Web.XmlTransform.XmlTransformableDocument
+        protected void XmlTransformableDocumentTest()
+        {
+            XmlTransformableDocument doc = new XmlTransformableDocument();
+        }
+    }
+
+    /// <summary>
+    ///  These are not testing the APIs per se, but other test combinations
+    /// </summary>
+    class VariousUnsafeCombinations
+    {
+        public void InsideTryCatch()
+        {
+            ConfigXmlDocument doc;
+            try
+            {
+                doc = new ConfigXmlDocument();
+                doc.XmlResolver = new XmlPreloadedResolver(); // Noncompliant
+            }
+            catch
+            {
+            }
+        }
+
+        public void InsideLoop()
+        {
+            ConfigXmlDocument doc;
+            while (true)
+            {
+                doc = new ConfigXmlDocument();
+                doc.XmlResolver = new XmlPreloadedResolver(); // Noncompliant
+            }
+        }
+
+        private void LocalFunction()
+        {
+            var doc = new ConfigXmlDocument();
+            LocalFunction();
+
+            void LocalFunction()
```

Comment:
We could also add a test with a lambda function which returns `new XmlPreloadedResolver()`